reactos/drivers/storage/storahci/storahci.c

```diff
@@ -76,60 +97,64 @@ BOOLEAN AhciPortInitialize(
  * @return
  * return TRUE if allocation was successful
  */
-BOOLEAN AhciAllocateResourceForAdapter(
-  __in      PAHCI_ADAPTER_EXTENSION             adapterExtension,
-  __in      PPORT_CONFIGURATION_INFORMATION     ConfigInfo
-)
+BOOLEAN
+AhciAllocateResourceForAdapter (
+    __in      PAHCI_ADAPTER_EXTENSION             adapterExtension,
+    __in      PPORT_CONFIGURATION_INFORMATION     ConfigInfo
+    )
 {
     PVOID portsExtension = NULL;
     PCHAR nonCachedExtension;
-    ULONG portCount, portImplemented, status, index, NCS, AlignedNCS, nonCachedExtensionSize, currentCount;
+    ULONG status, index, NCS, AlignedNCS;
+    ULONG portCount, portImplemented, nonCachedExtensionSize;
 
-    StorPortDebugPrint(0, "AhciAllocateResourceForAdapter()\n");
+    DebugPrint("AhciAllocateResourceForAdapter()\n");
 
-    // 3.1.1 NCS = CAP[12:08] -> Align
-    NCS = (adapterExtension->CAP & 0xF00) >> 8;
-    AlignedNCS = ((NCS/8) + 1) * 8;
+    NCS = AHCI_Global_Port_CAP_NCS(adapterExtension->CAP);
+    AlignedNCS = ROUND_UP(NCS, 8);
 
     // get port count -- Number of set bits in `adapterExtension->PortImplemented`
     portCount = 0;
     portImplemented = adapterExtension->PortImplemented;
-    while(portImplemented > 0)
+    while (portImplemented > 0)
     {
         portCount++;
-        portImplemented &= (portImplemented-1);
+        portImplemented &= (portImplemented - 1);
     }
-    StorPortDebugPrint(0, "\tPort Count: %d\n", portCount);
+
+    NT_ASSERT(portCount != 0);

```

Comment:
Is it even possible to get portCount > 32?
while loop just above this line is computing portCount. And that guarantees it to < 32.